src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/ECDsaCertificateExtensions.cs

```diff
@@ -64,5 +87,59 @@ private static bool HasECDsaKeyUsage(X509Certificate2 certificate)
             // considered valid for all usages, so we can use it for ECDSA.
             return true;
         }
+
+        private static bool IsSameKey(ECDsa a, ECDsa b)
+        {
+            ECParameters aParameters = a.ExportParameters(false);
+            ECParameters bParameters = b.ExportParameters(false);
+
+            if (aParameters.Curve.CurveType != bParameters.Curve.CurveType)
+                return false;
+            
+            if (!aParameters.Q.X.ContentsEqual(bParameters.Q.X) ||
+                !aParameters.Q.Y.ContentsEqual(bParameters.Q.Y))
+            {
+                return false;
+            }
+
+            ECCurve aCurve = aParameters.Curve;
+            ECCurve bCurve = bParameters.Curve;
+
+            if (aCurve.IsNamed)
+            {
+                // On Windows we care about FriendlyName, on Unix we care about Value
+                return (aCurve.Oid.Value == bCurve.Oid.Value && aCurve.Oid.FriendlyName == bCurve.Oid.FriendlyName);
```

Comment:
I suppose they could be equal even if there is a null\empty value in one curve but not in the other (e.g. Value) when the other side matches (e.g. FriendlyName), or perhaps the OId is not resolved yet. Not sure if this matters that much, however, as we could just assume consistent values.